From f4898c696436dc30cd0ca57a7c8f38e36c5b0ea5 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 21 Sep 2022 08:19:05 +0300
Subject: [PATCH] [SW-102429] gaudi2: ppboot: verify image validation

In case of secureboot_verify_image function exported
from arcboot, only check done in ppboot is for whether
function returned SECUREBOOT_VERIFY_FAILED. Correct
check for failure should be whether function returned
any error or not.

Change-Id: Ib1adc3722015a56258f38f04038fbd3bb10ac12a
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 489aa1b..cf27ce2 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -398,7 +398,7 @@
 	image_size -= SECUREBOOT_IMG_HEAD_FOOTER;
 	if (!preboot_security_fuse_override()) {
 		ret = secureboot_verify_image(image_size, SDRAM_BASE + IMG_LEN_SIZE);
-		if (ret == SECUREBOOT_VERIFY_FAILED)
+		if (ret <= SECUREBOOT_VERIFY_FAILED)
 			return ARCBOOT_BRINGUP_ERROR;
 	}
 
